components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2288,8 +2289,7 @@ public void makeAPIKeysForwardCompatible(API api) throws APIManagementException
         Collections.sort(sortedAPIs, comparator);
         for (int i = sortedAPIs.size() - 1; i >= 0; i--) {
             String oldVersion = sortedAPIs.get(i).getId().getVersion();
-            apiMgtDAO.makeKeysForwardCompatible(new ApiTypeWrapper(api), oldVersion
-            );
+            apiMgtDAO.makeKeysForwardCompatible(new ApiTypeWrapper(api), oldVersion, tenantId);
         }
```

Comment:
I don't think you have to pass the tenant Id , you could get the tenant id from the API provider using APIUtil.getTenantId(APIUtil.replaceEmailDomainBack(api.getId().getProviderName())

Reply:
resolved